src/engines/GPUEngine.ts

```diff
@@ -61,22 +102,37 @@ export default class GPUEngine {
   }
 
   // overspecified for compute shader
-  async execute(workgroupCount: [number, number, number]) {
+  async execute(video: HTMLVideoElement, ctx: GPUCanvasContext, workgroupCount: [number, number, number]) {
```

Comment:
why do we need the vide element as part of execute? i see that on L112 we call this.sendVideoData but that functionality might be better left to the consumer instead of forcing all consumers to pass a video?